src/dissection.jl

```diff
@@ -0,0 +1,152 @@
+using LightGraphs
+using DataStructures
+using StatsBase
+
+
+"""
+    separator!(g, subgraph_nodes, max_imbalance=0.6, max_nsample=20)
+Call flowcutter (with s and t random) on g and pick nodes at random in the cut matching the 60% imbalance
+heuristic to form a separator. Then split the graph g with respect to the separator.
+Parameter max_imbalance define the heurestic to choose a cut amoung all generated by
+flowcutter. Parameter max_nsample define the number of call to flowcutter.
+
+# Arguments
+- `g::SimpleGraph{Int64}`: the subgraph to run flowcutter.
+- `subgraph_nodes::Vector{Int64}`: a table where element at index i is corresponding label in the original root graph.
+- `max_imbalance::Float64 = 0.6` max imbalance to select cut from flowcutter result.
+- `max_nsample::Int64 = 20` max number of try to get a separator.
+# Return
+- `sep::Vector{Int64}`: the set of indices of the nodes in separator.
+- `split_parts::Array{Array{Int64,1},1}` several parts of the graph resulted from separation.
+"""
+function separator!(
+    g::SimpleGraph{Int64},
+    subgraph_nodes::Vector{Int64},
+    max_imbalance::Float64=0.6,
+    max_nsample::Int64=20,
+)::Tuple{Vector{Int64},Array{Array{Int64,1},1}}
+    @debug "------ separator! --------"
+
+    # run flowcutter many times and collect all of these cuts
+    cuts::Vector{Cut} = []
+    n = nv(g) + 2
+
+    dist = floyd_warshall_shortest_paths(g).dists
+    for i in 1:max_nsample
+        add_vertex!(g)
+        add_vertex!(g)
+        s, t = sample(1:length(subgraph_nodes), 2, replace=false)
+        push!.([cuts], flowcutter!(g, s, t, dist))
+        rem_vertex!(g, n)
+        rem_vertex!(g, n - 1)
+    end
+
+
+    # find best size imbalance for cuts
+    minsize = typemax(Int64)
+    minimbalance = typemax(Int64)
+    for c in cuts
+        size = c.size
+        imbalance = c.imbalance
+        expansion = c.expansion
+        if size < minsize
+            minsize = size
+        end
+        if imbalance < minimbalance
+            minimbalance = imbalance
+        end
+    end
+    # remove dominated cuts and cuts with more than max_imbalance (=0.6)
+    filter!(
+        c -> ((c.size == minsize) || (c.imbalance == minimbalance)) && (c.imbalance < max_imbalance),
+        cuts
+    )
+
+    # select cut with min expansion
+    cut = findmin(c -> c.expansion, cuts)[2]
+    sep = unique(map(a -> sample([a.first, a.second]), cut.arcs))
```

Comment:
This line gave me the following error when I ran the tests. It happened in the `test_dissection` test file when calling `nested_dissection` after the line printing "Testing nested_dissection with custom graph"
`LoadError: type Int64 has no field arcs`

Did you see the same error at any stage?

Reply:
I see the error once in an earlier version (when it was [1] and so gives the min expansion as an Int64) of the code but the [2] gives a Cut object and not an Int64 as far as I test